### DISCUSSION TITLE
Access to data_class object inside the formType buildForm method

### DISCUSSION THREAD

**Discussion by thgauthier** [symfony/symfony#44442] [Q&A] [answered]
hello everyone !
Is it possible to get the entity inside the formType buildForm method ?
I mean the entity witch is declared in the configureOptions method (data_class)
If yes how ?
The goal is to set the 'data' => to the return of the __toString method (or what ever) of the entity of an unmapped field.
With that unmapped field, I will do some staff client side with javascript.
Thx

**Comment by chapterjason** [comment, marked as answer]
You can access the option in `buildform` that is set in `configureOptions` like this:

```php
// src/Form/Type/TaskType.php
namespace App\Form\Type;

use Symfony\Component\Form\AbstractType;
use Symfony\Component\Form\Extension\Core\Type\DateType;
use Symfony\Component\Form\Extension\Core\Type\SubmitType;
use Symfony\Component\Form\Extension\Core\Type\TextType;
use Symfony\Component\Form\FormBuilderInterface;
use App\Entity\Task;

class TaskType extends AbstractType
{
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        var_dump($options['data_class']); // prints "App\Entity\Task"

        $builder
            ->add('task', TextType::class)
            ->add('dueDate', DateType::class)
            ->add('save', SubmitType::class)
        ;
    }

    public function configureOptions(OptionsResolver $resolver): void
    {
        $resolver->setDefaults([
            'data_class' => Task::class,
        ]);
    }
}
```

source: https://symfony.com/doc/current/forms.html#creating-form-classes

**Reply by thgauthier** [reply]
Thank very much for your answer.
I was looking for $options['data'] witch is the $task object itself.

**Reply by chapterjason** [reply]
Then just pass the model as an option on creation in the controller I guess. Don't forget to allow adding a new option in the configureOptions.

You can also use events, which I would prefer, depends on the detailed use case.
